src/api/VitessceConfig.js

```diff
@@ -0,0 +1,401 @@
+    this.view.w = w;
+    this.view.h = h;
+  }
+
+  /**
+   * @returns {object} This view as a JSON object.
+   */
+  toJSON() {
+    return this.view;
+  }
+}
+
+/**
+ * Class representing a horizontal concatenation of views.
+ */
+export class VitessceConfigViewHConcat {
+  constructor(views) {
+    this.views = views;
+  }
+}
+
+/**
+ * Class representing a vertical concatenation of views.
+ */
+export class VitessceConfigViewVConcat {
+  constructor(views) {
+    this.views = views;
+  }
+}
+
+/**
+ * A helper function to create a horizontal concatenation of views.
+ * @param  {...(VitessceConfigView|VitessceConfigViewHConcat|VitessceConfigViewVConcat)} views A
+ * variable number of views or view concatenations.
+ * @returns {VitessceConfigViewHConcat} A new horizontal view concatenation instance.
+ */
+export function hconcat(...views) {
+  const vcvhc = new VitessceConfigViewHConcat(views);
+  return vcvhc;
+}
+
+/**
+ * A helper function to create a vertical concatenation of views.
+ * @param  {...(VitessceConfigView|VitessceConfigViewHConcat|VitessceConfigViewVConcat)} views A
+ * variable number of views or view concatenations.
+ * @returns {VitessceConfigViewVConcat} A new vertical view concatenation instance.
+ */
+export function vconcat(...views) {
+  const vcvvc = new VitessceConfigViewVConcat(views);
+  return vcvvc;
+}
+
+/**
+ * Class representing a coordination scope in the coordination space.
+ */
+export class VitessceConfigCoordinationScope {
+  /**
+   * Construct a new coordination scope instance.
+   * @param {string} cType The coordination type for this coordination scope.
+   * @param {string} cScope The name of the coordination scope.
+   */
+  constructor(cType, cScope) {
+    this.cType = cType;
+    this.cScope = cScope;
+    this.cValue = null;
+  }
+
+  /**
+   * Set the coordination value of the coordination scope.
+   * @param {any} cValue The value to set.
+   * @returns {VitessceConfigCoordinationScope} This, to allow chaining.
+   */
+  setValue(cValue) {
+    this.cValue = cValue;
+    return this;
+  }
+}
+
+/**
+ * Class representing a Vitessce view config.
+ */
+export class VitessceConfig {
+  /**
+   * Construct a new view config instance.
+   * @param {string} name A name for the config. Optional.
+   * @param {string} description A description for the config. Optional.
+   */
+  constructor(name = '', description = '') {
+    this.config = {
+      version: '1.0.0',
+      name,
+      description,
+      datasets: [],
+      coordinationSpace: {},
+      layout: [],
+      initStrategy: 'auto',
+    };
+  }
+
+  /**
+   * Add a new dataset to the config.
+   * @param {string} name A name for the dataset. Optional.
+   * @param {string} uid Optional.
+   * @returns {VitessceConfigDataset} A new dataset instance.
+   */
+  addDataset(name = '', uid = null) {
+    const prevDatasetUids = this.config.datasets.map(d => d.dataset.uid);
+    const nextUid = (uid || getNextScope(prevDatasetUids));
```

Comment:
Could a `uid` be among the previous scopes?  Is this meant to allow the same scope to be reused across datasets?

Reply:
The `uid` is meant to be used internally for the `VitessceConfig.fromJSON`, not really for external users

Reply:
I will add a comment in the parameter JSDoc line to clarify this.